src/cli/commands/lint.ts

```diff
@@ -163,12 +169,17 @@ async function lint(name: string, flags: any, command: Lint, rules?: RuleCollect
   let results = [];
   try {
     const parsedResult: IParsedResult = {
-      source: resolve(process.cwd(), name),
+      source: targetUri,
       parsed: spec,
       getLocationForJsonPath,
     };
 
-    results = await spectral.run(parsedResult);
+    results = await spectral.run(parsedResult, {
+      resolve: {
+        authority: targetUri,
```

Comment:
Why is it called authority? It's not quite clicking for me yet.

Reply:
Authority is just piggybacking off of the `authority` option in https://github.com/stoplightio/json-ref-resolver. I created json-ref-resolver quite a while ago, and in hindsight it's not the best name for that property. I have an open ticket to make an update to json-ref-resolver to use https://github.com/microsoft/vscode-uri rather than urijs (to fix a bunch of issues with windows style file paths), and was thinking about leveraging that breaking release to also update some of the names.

We could decide on a name right now, and use that in spectral so that we don't have a breaking change here later? The authority option of json-ref-resolver is described here: https://github.com/stoplightio/json-ref-resolver#example-resolving-relative-remote-references-with-the-authority-option.

Basically, it's the location of the document currently being resolved, so that relative $refs inside of it can be effectively resolved (solution to #171).

Open to suggestions, here are some:

- parentUri
- documentUri
- uri
- $id (from json schema, "It declares a base URI against which $ref URIs are resolved."). This is basically exactly how we're using it.
- documentId

Reply:
OHHH i see! documentUri sounds great. Maybe we could switch to vscode-uri and do all the breaky things in one go?

Reply:
> I have an open ticket to make an update to json-ref-resolver to use https://github.com/microsoft/vscode-uri rather than urijs (to fix a bunch of issues with windows style file paths), and was thinking about leveraging that breaking release to also update some of the names.

:)